src/battle_controller_opponent.c

```diff
@@ -637,6 +617,184 @@ static void OpponentHandleChooseMove(u32 battler)
+    {SPECIES_PIKIPEK,        SPECIES_METAPOD},
+    {SPECIES_GUMSHOOS,       SPECIES_RATTATA},
+    {SPECIES_GUMSHOOS,       SPECIES_SKWOVET},
+    {SPECIES_LYCANROC,       SPECIES_DEERLING},
+    {SPECIES_MAREANIE,       SPECIES_PINCURCHIN},
+    {SPECIES_LURANTIS,       SPECIES_KRICKETOT},
+    {SPECIES_SALANDIT,       SPECIES_COMBEE},
+    {SPECIES_BRUXISH,        SPECIES_SHELLDER},
+    {SPECIES_DHELMISE,       SPECIES_WAILMER},
+    {SPECIES_GREEDENT,       SPECIES_BOUNSWEET},
+    {SPECIES_CORVIKNIGHT,    SPECIES_BUNNELBY},
+    {SPECIES_APPLETUN,       SPECIES_CUTIEFLY},
+    {SPECIES_CRAMORANT,      SPECIES_ARROKUDA},
+    {SPECIES_CENTISKORCH,    SPECIES_SCOLIPEDE},
+    {SPECIES_WATTREL,        SPECIES_ARROKUDA},
+    {SPECIES_VELUZA,         SPECIES_WIGLETT},
+    {SPECIES_GRAFAIAI,       SPECIES_SCATTERBUG},
+    // Rivals
+    {SPECIES_CUTIEFLY,       SPECIES_BUTTERFREE},
+    {SPECIES_SHIINOTIC,      SPECIES_PARASECT},
+    {SPECIES_URSARING,       SPECIES_PRIMEAPE},
+    {SPECIES_ROCKRUFF,       SPECIES_GROWLITHE},
+    {SPECIES_KINGDRA,        SPECIES_DRAGONITE},
+    {SPECIES_VIKAVOLT,       SPECIES_HERACROSS},
+    {SPECIES_DEWPIDER,       SPECIES_SURSKIT},
+    {SPECIES_BASCULIN,       SPECIES_CARVANHA},
+    {SPECIES_PASSIMIAN,      SPECIES_AMBIPOM},
+    {SPECIES_SQUAWKABILLY,   SPECIES_FLETCHLING},
+    {SPECIES_BRUXISH,        SPECIES_TOXAPEX},
+    {SPECIES_TEDDIURSA,      SPECIES_BEEDRILL},
+    {SPECIES_MURKROW,        SPECIES_MEOWTH},
+    {SPECIES_HAWLUCHA,       SPECIES_PRIMEAPE},
+    {SPECIES_VIKAVOLT,       SPECIES_PINSIR},
+    {SPECIES_LUMINEON,       SPECIES_LANTURN},
+    {SPECIES_CORVIKNIGHT,    SPECIES_SKARMORY},
+    {SPECIES_GABITE,         SPECIES_SABLEYE},
+    {SPECIES_GARCHOMP,       SPECIES_SALAMENCE},
+    {SPECIES_SIZZLIPEDE,     SPECIES_VENIPEDE},
+    {SPECIES_FRIGIBAX,       SPECIES_BERGMITE},
+    {SPECIES_FINIZEN,        SPECIES_BARRASKEWDA},
+    //Parasite               //Prey
+    {SPECIES_HYPNO,          SPECIES_KOMALA},
+    {SPECIES_ELEKID,         SPECIES_TOGEDEMARU},
+    {SPECIES_DWEBBLE,        SPECIES_ROGGENROLA},
+    {SPECIES_DWEBBLE,        SPECIES_ROLYCOLY},
+    {SPECIES_JOLTIK,         SPECIES_YAMPER},
+    {SPECIES_CUTIEFLY,       SPECIES_GOSSIFLEUR},
+    {SPECIES_TOGEDEMARU,     SPECIES_ELEKID},
+    {SPECIES_LECHONK,        SPECIES_APPLETUN},
+    {SPECIES_TINKATUFF,      SPECIES_PAWNIARD},
+    {SPECIES_TINKATUFF,      SPECIES_BISHARP},
+    {SPECIES_TINKATON,       SPECIES_CORVIKNIGHT},
+#endif
+};
+
+static bool32 CheckIfPartnerIsNaturalEnemy(u32 battler)
+{
+    u32 i;
+    u16 speciesAttacker = gBattleMons[battler].species;
+    u16 speciesTarget = gBattleMons[GetBattlerAtPosition(BATTLE_PARTNER(battler))].species;
+
+    if ((speciesAttacker == SPECIES_CARBINK) && (speciesTarget == SPECIES_SABLEYE))
+        return FALSE;
+
+    if ((speciesAttacker == SPECIES_CORSOLA) && (speciesTarget == SPECIES_MAREANIE))
+        return FALSE;
```

Comment:
These 2 fall into the "predator vs prey" category, but all other mon that you classified as such don't behave this way.
I would recommend having separate arrays for the different types of interactions.

Reply:
https://github.com/rh-hideout/pokeemerald-expansion/pull/3622/commits/1144b9c250f8f2f2e7ee19e3e5333c3647dedd07